pkg/segment/query/iqr/intermediateQueryResult_test.go

```diff
@@ -431,6 +431,73 @@ func Test_DiscardAfter(t *testing.T) {
 	assert.Equal(t, 1, iqr.NumberOfRecords())
 }
 
+func Test_MergeWithoutRRCIQRIntoRRCIQR(t *testing.T) {
+	rrcIqr := NewIQR(0)
+	segKeyInfo1 := utils.SegKeyInfo{
+		SegKeyEnc: 1,
+	}
+	encodingToSegKey := map[uint16]string{1: "segKey1"}
+	rrcs := []*utils.RecordResultContainer{
+		{SegKeyInfo: segKeyInfo1, BlockNum: 1, RecordNum: 1},
+		{SegKeyInfo: segKeyInfo1, BlockNum: 1, RecordNum: 2},
+		{SegKeyInfo: segKeyInfo1, BlockNum: 1, RecordNum: 3},
+	}
+
+	err := rrcIqr.AppendRRCs(rrcs, encodingToSegKey)
+	assert.NoError(t, err)
+
+	withoutRRCIqr := NewIQR(0)
+	knownValues := map[string][]utils.CValueEnclosure{
+		"col1": {
+			utils.CValueEnclosure{Dtype: utils.SS_DT_STRING, CVal: "a1"},
+			utils.CValueEnclosure{Dtype: utils.SS_DT_STRING, CVal: "a2"},
+			utils.CValueEnclosure{Dtype: utils.SS_DT_STRING, CVal: "a3"},
+		},
+		"col2": {
+			utils.CValueEnclosure{Dtype: utils.SS_DT_STRING, CVal: "b1"},
+			utils.CValueEnclosure{Dtype: utils.SS_DT_STRING, CVal: "b2"},
+			utils.CValueEnclosure{Dtype: utils.SS_DT_STRING, CVal: "b3"},
+		},
+	}
+	err = withoutRRCIqr.AppendKnownValues(knownValues)
+	assert.NoError(t, err)
+	numGeneratedCol := withoutRRCIqr.NumberOfRecords()
+	assert.Equal(t, 3, numGeneratedCol)
+
+	convertedRRCIQR, err := withoutRRCIqr.getRRCIQR()
+	assert.NoError(t, err)
+	assert.Equal(t, withRRCs, convertedRRCIQR.mode)
+	assert.Equal(t, 3, len(convertedRRCIQR.rrcs))
+	assert.Equal(t, knownValues, convertedRRCIQR.knownValues)
+	assert.Equal(t, 3, convertedRRCIQR.NumberOfRecords())
+
+	err = rrcIqr.Append(withoutRRCIqr)
+	assert.Nil(t, err)
+	assert.Equal(t, withRRCs, rrcIqr.mode)
+	assert.Equal(t, 6, rrcIqr.NumberOfRecords())
+
+	nilCValEnc := utils.CValueEnclosure{Dtype: utils.SS_DT_BACKFILL, CVal: nil}
+
+	for col, values := range rrcIqr.knownValues {
+		for i, value := range values {
+			if i < numGeneratedCol {
+				assert.Equal(t, nilCValEnc, value)
+			} else {
+				assert.Equal(t, knownValues[col][i-numGeneratedCol], value)
+			}
+		}
+	}
+
+	knownValues["col1"] = knownValues["col1"][1:]
+	newIQR := NewIQR(0)
+	err = newIQR.AppendKnownValues(knownValues)
+	assert.Error(t, err)
```

Comment:
I might be misunderstanding. But this looks like you're just validating `AppendKnownValues` since the two columns have different numbers of records. So it's really unrelated to your test `Test_MergeWithoutRRCIQRIntoRRCIQR`, so it should be moved to a different test.